File: app/src/main/java/com/moh/mhealth/pages/Diagnostics1.java
package com.moh.mhealth.pages;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.NumberPicker;
import android.widget.TextView;

import com.moh.mhealth.Global_Helper;
import com.moh.mhealth.Header;
import com.moh.mhealth.Patient;
import com.moh.mhealth.R;

import org.w3c.dom.Text;

// Set weight, height, and temperature
public class Diagnostics1 extends AppCompatActivity implements Header {

    NumberPicker height1Picker, height2Picker;

    EditText tempField, weightField;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_diagnostics1);


        height1Picker = (NumberPicker) findViewById(R.id.diag1_height1_num);
        height2Picker = (NumberPicker) findViewById(R.id.diag1_height2_num);

        height1Picker.setMinValue(0);
        height1Picker.setMaxValue(250);
        height1Picker.setValue(175);

        tempField = (EditText) findViewById(R.id.diag1_temp_num);
        weightField = (EditText) findViewById(R.id.diag1_weight_num);

        if (! Global_Helper.isMetric()) {
            setImperial();
        }

        loadPatientData();

        Button nextBtn = (Button) findViewById(R.id.diag1_next);

        nextBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                updatePatientData();

                startActivity(new Intent(getApplicationContext(), Global_Helper.nextDiag()));
            }
        });
    }

    private void setImperial() {
        // Set kg to lbs
        ((TextView) findViewById(R.id.diag1_weight_unit_text)).setText("lbs");

        // Set cm to ft
        ((TextView) findViewById(R.id.diag1_height1_unit_text)).setText("ft");

        // Change max values of the two number pickers
        height1Picker.setMaxValue(10);
        height1Picker.setValue(5);
        height2Picker.setMaxValue(11);

        // Show the inches layout
        ((LinearLayout) findViewById(R.id.diag1_height2)).setVisibility(View.VISIBLE);

        // Set C to F
        ((TextView) findViewById(R.id.diag1_temp_unit_text)).setText("\u00B0F");
    }

    private void updatePatientData() {
        double weight = Global_Helper.getDoubleFromEditText(weightField);

        int height1 = height1Picker.getValue();

        double temp = Global_Helper.getDoubleFromEditText(tempField);

        if (! Global_Helper.isMetric()) { // Gonna have to convert
            Global_Helper.addDiag1Imperial(weight,
                                           height1,
                                           height2Picker.getValue(),
                                           temp);
        } else {
            Global_Helper.addDiag1Metric(weight,
                                         height1,
                                         temp);
        }
    }

    private void loadPatientData() {
        Patient patient = Global_Helper.getCurrentPatient();

        double weight = patient.getWeight();
        int height = patient.getHeight();
        double temp = patient.getTemp();

        if (! Global_Helper.isMetric()) { // Gonna have to convert
            if (weight >= 0) {
                weight = Global_Helper.getKgToLb(weight);
            }

            /*

            125.9
            6' 8''
            102.5
             */
            if (height > 0) {
                int heightInches = (int) Math.round(Global_Helper.getCmToInch(height));

                height = heightInches / 12;

                height2Picker.setValue(heightInches % 12);
            }

            if (temp >= 0) {
                temp = Global_Helper.getCToF(temp);
            }
        }

        TextView patient_name = (TextView) findViewById(R.id.header_name);
        patient_name.setText(patient.getName());

        if (weight >= 0) {
            weightField.setText(weight + "");
        }
        height1Picker.setValue(height);

        if (temp >= 0) {
            tempField.setText(temp + "");
        }
    }


    @Override
    public void cancel(View view) {
        startActivity(new Intent(getApplicationContext(), Global_Helper.endPatient()));
    }

    @Override
    public void moveBack(View view) {
        startActivity(new Intent(getApplicationContext(), Global_Helper.prevDiag()));
    }
}

File: app/src/main/java/com/moh/mhealth/pages/Diagnostics1.kt
package com.moh.mhealth.pages

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.moh.mhealth.GlobalHelper
import com.moh.mhealth.Header
import com.moh.mhealth.R
import kotlin.math.roundToInt

// Set weight, height, and temperature
class Diagnostics1 : AppCompatActivity(), Header {
    private var height1Picker: NumberPicker? = null
    private var height2Picker: NumberPicker? = null
    private var tempField: EditText? = null
    private var weightField: EditText? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_diagnostics1)
        height1Picker = findViewById<View>(R.id.diag1_height1_num) as NumberPicker
        height2Picker = findViewById<View>(R.id.diag1_height2_num) as NumberPicker
        height1Picker!!.minValue = 0
        height1Picker!!.maxValue = 250
        height1Picker!!.value = 175
        tempField = findViewById<View>(R.id.diag1_temp_num) as EditText
        weightField = findViewById<View>(R.id.diag1_weight_num) as EditText
        if (!GlobalHelper.isMetric) {
            setImperial()
        }
        loadPatientData()
        val nextBtn = findViewById<View>(R.id.diag1_next) as Button
        nextBtn.setOnClickListener {
            updatePatientData()
            startActivity(Intent(applicationContext, GlobalHelper.nextDiag()))
        }
    }

    @SuppressLint("SetTextI18n")
    private fun setImperial() {
        // Set kg to lbs
        (findViewById<View>(R.id.diag1_weight_unit_text) as TextView).text = "lbs"

        // Set cm to ft
        (findViewById<View>(R.id.diag1_height1_unit_text) as TextView).text = "ft"

        // Change max values of the two number pickers
        height1Picker!!.maxValue = 10
        height1Picker!!.value = 5
        height2Picker!!.maxValue = 11

        // Show the inches layout
        (findViewById<View>(R.id.diag1_height2) as LinearLayout).visibility =
            View.VISIBLE

        // Set C to F
        (findViewById<View>(R.id.diag1_temp_unit_text) as TextView).text = "\u00B0F"
    }

    private fun updatePatientData() {
        val weight = GlobalHelper.getDoubleFromEditText(weightField)
        val height1 = height1Picker!!.value
        val temp = GlobalHelper.getDoubleFromEditText(tempField)
        if (!GlobalHelper.isMetric) { // Gonna have to convert
            GlobalHelper.addDiag1Imperial(
                weight,
                height1,
                height2Picker!!.value,
                temp
            )
        } else {
            GlobalHelper.addDiag1Metric(
                weight,
                height1,
                temp
            )
        }
    }

    private fun loadPatientData() {
        val patient = GlobalHelper.currentPatient!!
        var weight = patient.weight
        var height = patient.height
        var temp = patient.temp
        if (!GlobalHelper.isMetric) { // Gonna have to convert
            if (weight != null) {
                weight = GlobalHelper.getKgToLb(weight)
            }
            if (height != null) {
                val heightInches = GlobalHelper.getCmToInch(height.toDouble()).roundToInt()
                height = heightInches / 12
                height2Picker!!.value = heightInches % 12
            }
            if (temp != null) {
                temp = GlobalHelper.getCToF(temp)
            }
        }
        val patientName = findViewById<View>(R.id.header_name) as TextView
        patientName.text = patient.name
        if (weight != null) {
            weightField!!.setText(weight.toString())
        }
        if (height != null) {
            height1Picker!!.value = height
        }
        if (temp != null) {
            tempField!!.setText(temp.toString())
        }
    }

    override fun cancel(view: View) {
        startActivity(Intent(applicationContext, GlobalHelper.endPatient()))
    }

    override fun moveBack(view: View) {
        startActivity(Intent(applicationContext, GlobalHelper.prevDiag()))
    }
}